### PR TITLE
[React Native] Update React Native version and ScrollView typings

### PR DESCRIPTION
**Pull request by gyzerok**
### Template

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/scrollview.html
- [x] Increase the version number in the header if appropriate.

### About React Native version update

I understand that there are probably other breaking changes introduced from 0.37 to 0.42. The thing is that people use "old" version typings with new version anyway. It feels unfeasible to update everything for one person. So by updating this number I expect people to have some issues, but I expect them to come here and make appropriate changes, so typings becoming better and everyone wins after all.

### CODE REVIEW

File: react-native/index.d.ts [hunk 1]
```diff
@@ -5524,24 +5534,24 @@ declare module "react" {
...
-	
+        keyboardShouldPersistTaps?: boolean | 'always' | 'never' | 'handled'
+
+        /**
+         * Called when scrollable content view of the ScrollView changes.
+         * Handler function is passed the content width and content height as parameters: (contentWidth, contentHeight)
+         * It's implemented using onLayout handler attached to the content container which this ScrollView renders.
+         *
+         */
+        onContentSizeChange?: (w: number, h: number) => void
```
Comment (gyzerok) [inline react-native/index.d.ts]: Here it was just incorrect indentation.

**Comment by gyzerok** [comment]
If we can get this PR with updated version through, I would love to go through the docs and update as much definitions as I can by myself.

File: react-native/index.d.ts [hunk 2]
```diff
@@ -5414,11 +5414,6 @@ declare module "react" {
         onScrollAnimationEnd?: () => void
 
         /**
-         * When false, the content does not scroll. The default value is true
-         */
-        scrollEnabled?: boolean // true
```
Comment (gyzerok) [inline react-native/index.d.ts]: According to docs it is not specific to iOS only

**Comment by gyzerok** [comment]
@johnnyreilly not sure why bot haven't mention someone specific. Could you, please take a look?

**Comment by johnnyreilly** [comment]
Thanks!

File: react-native/index.d.ts [hunk 3]
```diff
@@ -5452,7 +5447,7 @@ declare module "react" {
          * This can be used for paginating through children that have lengths smaller than the scroll view.
          * Used in combination with snapToAlignment.
          */
-        snapToInterval?: number
+        snapToInterval?: number[]
```
Comment (jasonhealy) [inline react-native/index.d.ts]: Shouldn't this be a number? An Array here break the build.
https://github.com/facebook/react-native/blob/e974798656eb811036970b72f064dc8a9d10b229/Libraries/Components/ScrollView/ScrollView.js#L309

**Pull request merged**